airbyte_cdk/cli/build/_run.py

```diff
@@ -0,0 +1,423 @@
+
+logger = logging.getLogger("airbyte-cdk.cli.build")
+
+
+def set_up_logging(verbose: bool = False) -> None:
+    """Set up logging configuration."""
+    log_level = logging.DEBUG if verbose else logging.INFO
+    logging.basicConfig(
+        level=log_level,
+        format="%(asctime)s [%(levelname)s] %(message)s",
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+
+
+def parse_args(args: List[str]) -> argparse.Namespace:
+    """Parse command line arguments for the build command."""
+    parser = argparse.ArgumentParser(
+        description="Build connector Docker images using the host Docker daemon"
+    )
+    parser.add_argument("connector_dir", type=str, help="Path to the connector directory")
+    parser.add_argument(
+        "--tag", type=str, default="dev", help="Tag to apply to the built image (default: dev)"
+    )
+    parser.add_argument(
+        "--no-verify", action="store_true", help="Skip verification of the built image"
+    )
+    parser.add_argument("--verbose", "-v", action="store_true", help="Enable verbose logging")
+
+    return parser.parse_args(args)
+
+
+def read_metadata(connector_dir: Path) -> ConnectorMetadata:
+    """Read and parse connector metadata from metadata.yaml.
+
+    Args:
+        connector_dir: Path to the connector directory.
+
+    Returns:
+        The parsed connector metadata.
+
+    Raises:
+        FileNotFoundError: If the metadata.yaml file doesn't exist.
+        ValueError: If the metadata is invalid.
+    """
+    metadata_path = connector_dir / "metadata.yaml"
+    if not metadata_path.exists():
+        raise FileNotFoundError(f"Metadata file not found: {metadata_path}")
+
+    metadata_content = metadata_path.read_text()
+    metadata_dict = yaml.safe_load(metadata_content)
+
+    if not metadata_dict or "data" not in metadata_dict:
+        raise ValueError("Invalid metadata format: missing 'data' field")
+
+    metadata_file = MetadataFile.model_validate(metadata_dict)
+    return metadata_file.data
+
+
+def infer_connector_language(metadata: ConnectorMetadata, connector_dir: Path) -> ConnectorLanguage:
+    """Infer the connector language from metadata and the file structure.
+
+    Args:
+        metadata: The connector metadata.
+        connector_dir: Path to the connector directory.
+
+    Returns:
+        The inferred connector language.
+    """
+    if metadata.language is not None:
+        return metadata.language
+
+    if (connector_dir / "setup.py").exists() or (connector_dir / "pyproject.toml").exists():
+        return ConnectorLanguage.PYTHON
+
+    if (connector_dir / "build.gradle").exists():
+        return ConnectorLanguage.JAVA
+
+    if any((connector_dir / f).exists() for f in ["manifest.yaml", "spec.yaml", "spec.json"]):
+        return ConnectorLanguage.LOW_CODE
+
+    logger.warning("Could not determine connector language, using UNKNOWN.")
+    return ConnectorLanguage.UNKNOWN
```

Comment:
let's remove this. Let the caller just inspect the `ConnectorMetadata` object and fail if the language isn't declared.